### PR TITLE
[RSDK-746] Return the power percent when checking if a motor is powered

### PR DESCRIPTION
**Pull request by penguinland**
I'm a little nervous that there are so many different implementations of this feature across different motors, and maybe there was a simpler or more consistent thing to do. The general idea here is that the `Motor` interface's `IsPowered` now returns the power level in addition to a boolean and a possible error. This is then used to populate the new `powerPct` field in the `IsPoweredResponse` protobuf.

The changes to go.mod and go.sum are intended to pick up https://github.com/viamrobotics/api/pull/70. 

Most of the motors didn't have an obvious way to get the power percent, so I started storing that in the relevant structs, updating that on a `SetPower` call, and returning whatever had been set. 

Despite all the tests passing, I have low confidence that I've done this right. What if I forgot to update the stored percent in a certain place? What if I updated it but a subsequent error means it's not actually set in the hardware? What if this PR is so big that everyone's eyes glaze over before they reach the end and we all miss an important bug? 

What can I do to gain confidence that this change was done correctly?

### CODE REVIEW

**Comment by penguinland** [comment]
This PR is larger than I'd prefer, but I didn't see an obvious way to split it up into smaller pieces. Next time, what should I do differently?

**Comment by penguinland** [comment]
:eyes: Running `make go-test` passes, but Github can find tests that have type mismatches and need to be updated! What's the preferred way to run all the tests, including the ones that `make go-test` skips?

File: components/motor/motor_test.go [hunk 1]
```diff
@@ -153,8 +153,8 @@ func TestCreateStatus(t *testing.T) {
 	status := &pb.Status{IsPowered: true, PositionReporting: true, Position: 7.7, IsMoving: true}
 
 	injectMotor := &inject.LocalMotor{}
-	injectMotor.IsPoweredFunc = func(ctx context.Context, extra map[string]interface{}) (bool, error) {
-		return status.IsPowered, nil
+	injectMotor.IsPoweredFunc = func(ctx context.Context, extra map[string]interface{}) (bool, float64, error) {
+		return status.IsPowered, 2.0 /* TODO: update this if it gets used */, nil
```
Comment (npmenard) [inline components/motor/motor_test.go:157]: why 2.0?

**Review by npmenard — APPROVED** [review]
LGTM! just left two comments most important IMO is whether or not we need to collect the powerPct in the collector

File: components/motor/motor_test.go (hunk 1 above)
Comment (penguinland) [reply components/motor/motor_test.go:157]: Turns out this value is completely unused. I wanted to make it `nil` but Go didn't like that for a float. So, I made it a nonsensical value, in the hopes that someone changes it to something sane if it ever becomes used (and fails some test if that doesn't happen). 

...in hindsight, setting it to 1.0 (i.e., full power) is probably reasonable for any test that uses it in the future. I'll go make that change.

File: components/motor/collectors.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func newIsPoweredCollector(resource interface{}, params data.CollectorParams) (d
 	}
 
 	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
-		v, err := motor.IsPowered(ctx, nil)
+		v, _, err := motor.IsPowered(ctx, nil)
```
[the remark this answers is not in the document]
Comment (penguinland) [reply components/motor/collectors.go:61]: I plead ignorance! What's a collector? I didn't understand what this file was for, so made the minimum change that wouldn't affect any behavior.

[the remark this answers is not in the document]
Comment (penguinland) [reply components/motor/collectors.go]: Well, I've taken a stab at updating the collector to track the new value, too. LMK if I should have done something different.

**Pull request merged**